gson-problem/src/main/java/org/zalando/problem/ProblemAdapterFactory.java

```diff
@@ -0,0 +1,138 @@
+package org.zalando.problem;
+
+import com.google.gson.*;
+import com.google.gson.internal.Streams;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.*;
+
+/**
+ * Problem {@link TypeAdapterFactory}.
+ */
+public class ProblemAdapterFactory implements TypeAdapterFactory {
+
+    private static final Set<Class<? extends Problem>> BASE;
+
+    static {
+        BASE = new HashSet<>();
+        BASE.add(Problem.class);
+        BASE.add(Exceptional.class);
+        BASE.add(ThrowableProblem.class);
+        BASE.add(AbstractThrowableProblem.class);
+        BASE.add(DefaultProblem.class);
+    }
+
+    private final boolean stacktrace;
+    private final Map<URI, Class<? extends Problem>> subtypes;
+    private final StatusTypeAdapter statusAdapter;
+
+    public ProblemAdapterFactory() {
+        this(false);
+    }
+
+    public ProblemAdapterFactory(boolean stacktrace) {
```

Comment:
I'd liek to get rid of this in favor of a `withStackTraces()` method:

https://github.com/zalando/problem/blob/05ae747b84e7f1c7a604975626eb8ea02cedd734/jackson-datatype-problem/src/main/java/org/zalando/problem/ProblemModule.java#L95-L101